op-supervisor/supervisor/backend/syncsrc/iface.go

```diff
@@ -0,0 +1,29 @@
+package syncsrc
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+type SyncSourceCollection interface {
+	Load(ctx context.Context, logger log.Logger) ([]SyncSourceSetup, error)
+	Check() error
+}
+
+type SyncSourceSetup interface {
+	Setup(ctx context.Context, logger log.Logger) (SyncSource, error)
+}
+
+type SyncSource interface {
+	BlockRefByNumber(ctx context.Context, number uint64) (eth.BlockRef, error)
+	FetchReceipts(ctx context.Context, blockHash common.Hash) (gethtypes.Receipts, error)
+	ChainID(ctx context.Context) (types.ChainID, error)
+	// String identifies the sync source
+	String() string
+}
```

Comment:
Just curious if you think this would be the place to attach the derivation command APIs. That's what it seems like to me at least.

Reply:
(spoke offline about this -- yes)